app/services/participants/base.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require "abstract_interface"
+
+module Participants
+  class Base
+    include ProfileAttributes
+    include AbstractInterface
+    implement_class_method :valid_courses
+
+    class << self
+      def call(params:)
+        new(params: params).call
+      end
+    end
+
+    def call
+      unless valid?
+        raise ActionController::ParameterMissing, errors.map(&:message)
+      end
+
+      business_validation!
+      action!
+    end
+
+    def business_validation!
+      validate_provider!
+    end
```

Comment:
`business_validation` is never overwritten, and all it does is one line. Can we inline it?

Reply:
True. I changed it and then folded it into a standard validation for the inherited class. I'll inline.